addon/utils/locale.js

```diff
@@ -13,6 +15,10 @@ export default class Locale {
   //  3. Default `rtl` to `false`
   //  4. Ensure `pluralForm` is defined
   constructor(id, container) {
+    if (initializedLocales[id]) {
+      return initializedLocales[id];
+    }
```

Comment:
Argh. You're completely right. Unfortunately, actually doing this causes adding translations at runtime to stop working. So I guess this method of caching won't work after all. I'm really not sure how to handle this correctly, the architecture is hard for me to follow... should I just take it out for now? I can personally live with the sub-optimal performance, since my use case is not very intensive (this may be a case of premature optimization).